certificateshop-logic/src/main/java/com/epam/esm/configuration/LogicConfig.java

```diff
@@ -0,0 +1,99 @@
+        implements WebMvcConfigurer {
+    private static final String DRIVER_PROPERTY_NAME = "JDBC_DRIVER";
+    private static final String DB_MYSQL_PATH_PROPERTY_NAME = "DB_MYSQL_PATH";
+    private static final String DB_SERVER_PROPERTY_NAME = "DB_SERVER";
+    private static final String DB_PORT_PROPERTY_NAME = "DB_PORT";
+    private static final String DB_NAME_PROPERTY_NAME = "DB_NAME";
+    private static final String DB_USER_PROPERTY_NAME = "DB_USER";
+    private static final String DB_PASSWORD_PROPERTY_NAME = "DB_PASSWORD";
+    private static final String DB_MAX_CONNECTIONS_PROPERTY_NAME = "DB_MAX_CONNECTIONS";
+
+    List<Locale> LOCALES = Arrays.asList(
+            new Locale("en"),
+            new Locale("de"),
+            new Locale("ru"));
+
+    private final Environment environment;
+
+    @Autowired
+    public LogicConfig(Environment environment) {
+        this.environment = environment;
+    }
+
+    @Profile("test")
+    @Bean(destroyMethod = "shutdown")
+    public EmbeddedDatabase dataSourceTest() {
+        return new EmbeddedDatabaseBuilder()
+                .setType(EmbeddedDatabaseType.H2)
+                .addScript("sql/create_db.sql")
+                .addScript("sql/insert_data.sql")
+                .build();
+    }
+
+    @Profile("dev")
+    @Bean
+    public DataSource dataSource() {
+        HikariConfig config = new HikariConfig();
+        config.setDriverClassName(environment.getRequiredProperty(DRIVER_PROPERTY_NAME));
+        config.setJdbcUrl(environment.getRequiredProperty(DB_MYSQL_PATH_PROPERTY_NAME)
+                + environment.getRequiredProperty(DB_SERVER_PROPERTY_NAME)
+                + ":"
+                + environment.getRequiredProperty(DB_PORT_PROPERTY_NAME)
+                + "/"
+                + environment.getRequiredProperty(DB_NAME_PROPERTY_NAME));
+        config.setUsername(environment.getRequiredProperty(DB_USER_PROPERTY_NAME));
+        config.setPassword(environment.getRequiredProperty(DB_PASSWORD_PROPERTY_NAME));
+        config.setMaximumPoolSize(environment.getProperty(DB_MAX_CONNECTIONS_PROPERTY_NAME, Integer.class, 30));
+        return new HikariDataSource(config);
+    }
+
+    @Override
+    public Locale resolveLocale(HttpServletRequest request) {
+        String headerLang = request.getHeader("Accept-Language");
+        return headerLang == null || headerLang.isEmpty()
+                ? Locale.getDefault()
+                : Locale.lookup(Locale.LanguageRange.parse(headerLang), LOCALES);
+    }
+
+    @Bean
+    public ResourceBundleMessageSource messageSource() {
+        ResourceBundleMessageSource rs = new ResourceBundleMessageSource();
+        rs.setBasename("messages");
+        rs.setDefaultEncoding("UTF-8");
+        rs.setUseCodeAsDefaultMessage(true);
+        return rs;
+    }
+
+    @Bean
+    public JdbcTemplate jdbcTemplate(DataSource dataSource) {
+        return new JdbcTemplate(dataSource);
+    }
+}
```

Comment:
Due to code convention good practice is to add blank lines at the end of java class. To do that automatically you can add google-java-format plugin, enable it and use CTRL+ALT+L